utils/common/src/test/kotlin/ProcessCaptureTest.kt

```diff
@@ -36,6 +37,21 @@ class ProcessCaptureTest : StringSpec({
         proc.stdout.trimEnd() shouldBe "This is some path: /foo/bar"
     }
 
+    "Only allowed environment variables should be passed" {
+        val env = mapOf("DB_USER" to "scott", "DB_PASSWORD" to "tiger", "DB_CONN" to "my-db.example.org")
+
+        withEnvironment(env) {
+            val proc = if (Os.isWindows) {
+                ProcessCapture("cmd.exe", "/c", "echo %DB_USER%:%DB_PASSWORD%.%DB_CONN%")
```

Comment:
Just wondering, why are you using `.` as a separator before the `DB_CONN`? For a more realistic example, shouldn't it be `@`?

Reply:
Yes, but the "@" character could be problematic when it is interpreted by the shell. I did not want to mess around with escaping syntax, since for the test it is only relevant that some of the values are replaced while others are not.